docs/designs/conceptual-versioning.md

```diff
@@ -0,0 +1,512 @@
+monikerDefinitionUrl: "https://api.docs.com/monikers/"
+redirections:
+    articles/v1.0/old/a: /articles/a?view=netcore-1.0
+    articles/v2.0/old/a: /articles/a
+```
+
+### 3 Output
+
+#### 3.1 Output file path
+
+For dynamic, the output path shares the same schema:
+
+`{output-dir}/{locale}?/{group-{monikerListHash}}?/{site-path}`
+
+```txt
+      locale monikerListHash           site-path
+      |-^-| |------^-----| |----------------^----------------|
+_site/en-us/group-01ddf122/dotnet/api/system.string/index.html
+```
+
+> `?` means optional. When the file have no version, the output path will be `{output-dir}/{locale}?/{site-path}`  
+> `monikerListHash` is the first 8 characters of the hash of this file's final moniker list, joined by whitespace.
+
+#### 3.2 Output content
+
+- Content of conceptual file's output(`*.raw.page.json`)
+
+```json
+{
+    ...,
+    "monikers":
+    [
+        "moniker1",
+        ...
+    ]
+    ...,
+}
+```
+
+- Content for redirection file(`*.raw.page.json`)
+
+```json
+{
+  "outputRootRelativePath": "../",
+  "rawMetadata": {
+    "locale": "en-us",
+    "redirect_url": "{redirectURL}",
+    "monikers":
+        [
+            "moniker1",
+            ...
+        ]
+  },
+  "themesRelativePathToOutputRoot": "_themes/"
+}
+```
+
+- Content of Toc file's output(`toc.json`)
+
+```json
+{
+    "items": [
+        {
+            "toc_title": "title",
+            "monikers": [
+                "moniker1",
+                ...
+            ],
+            "href": "{href}"
+            "children": [...]
+        }
+    ...,
+    ]
+    "metadata": {
+        "monikers": [
+            "moniker1",
+            ...
+        ]
+    }
+    ...
+}
+```
+
+- Content for `.manifest.json`
+
+    ##### Ideal output
+
+    ```json
+    {
+        "files":[
+            {
+                "siteUrl": "/{SitePath}",
+                "outputPath": "{outputPath}",
+                "sourcePath": "{sourcePath}",
+                "monikers": [
+                    "moniker1",
+                    ...
+                ]
+            },
+        ]
+    }
+    ```
+
+    ##### Legacy output
+
+    ```json
+    {
+        "groups":{
+            "groupid" : [
+                "moniker1",
+                ...
+            ],
+            ...
+        },
+        "files":[
+            {
+                "siteUrl": "/{SitePath}",
+                "outputPath": "{outputPath}",
+                "sourcePath": "{sourcePath}",
+                "group": "{groupid}"
+            },
+        ]
+    }
+    ```
+
+    > `groupid` is the first 8 characters of the hash of the monikers joined by whitespace.
+
+### 4 Implementation
+
+#### 4.1 MonikerRange evaluation
+
+Moniker range itself cannot be interpreted, but within an ordered moniker list, moniker range can be simply evaluated to a list of monikers.
+
+we can get an ordered moniker list by the API set in config as `monikerDefinitionUrl`, the response structure should be:
+
+```json
```

Comment:
create a container object, remove order, remove `name`, use snake_casing to be consistent with others?:

```json
{
  "monikers": [
    {
       "moniker": xxx
       "product": xxx
       "product_family": xxx
       "platform": xxx
       "display_name": xxx  
    }
   ]
}
```